### ISSUE TITLE
Fl_Terminal has some issues with tabs if it is created with 0 size

### ISSUE THREAD

**Issue by gnuwimp** [fltk/fltk#882]
**Describe the bug**
Fl_Terminal has some issues with tabs if it is created with 0 width and height.
I usually create all child widgets with size set to 0 and resizes them dynamically later on.

**To Reproduce**
```
// g++ -o terminal.exe terminal.cpp `fltk-config --cxxflags --ldflags`

#include <FL/Fl.H>
#include <FL/Fl_Window.H>
#include <FL/Fl_Terminal.H>

int main() {
    Fl_Window* w = new Fl_Window(100, 100, 600, 400, "terminal.cpp");
//    Fl_Terminal* t = new Fl_Terminal(0, 0, 600, 400);
    Fl_Terminal* t = new Fl_Terminal(0, 0, 0, 0);

    t->resize(0, 0, 600, 400);
    Fl::check();
    t->reset_terminal();
    w->resizable(t);
    w->show();
    t->append("Hello\tWorld\n");
    t->append("\tHello World\n");
    return Fl::run();
}

```
**Screenshots**
![terminal](https://github.com/fltk/fltk/assets/18072526/cda73349-4357-4b78-b18a-b95f7ceb5f0c)

**FLTK Version**
Current 1.4.0 d237d186ae1db050ce07a6e39cc2fb4d455f150a.

**FLTK Configure / Build Options**
./configure --enable-use_std

**Operating System / Platform:**
Xubuntu 23.10

**Comment by erco77** [comment]
Mmm, sounds like the terminal reset code isn't reconstructing the tabs. Will look at this later today.

**Comment by erco77** [comment]
OK, fixed in c9534460f.
Thanks for the report -- please reopen if there's still a problem!
